csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -30,15 +29,40 @@ void PropagateShardingsPass::runPass(Fusion* fusion) {
 
     // Note: Tvs without a mesh are assumed to have no manual sharding
     // annotation and are sharded like the first producer Tv.
+    const auto& outputs = ir_utils::filterByType<TensorView>(expr->outputs());
     std::vector<TensorView*> outputs_without_mesh;
-    for (TensorView* tv : outputs) {
+    for (auto* tv : outputs) {
       if (!tv->hasDeviceMesh()) {
         outputs_without_mesh.push_back(tv);
       }
     }
     shardAllLike(input_with_mesh, outputs_without_mesh);
   }
 
+  // Back-propagate device meshes. This is needed in addition to the forward
```

Comment:
This will cause some issues where a DID parallelization needed to be back-propagated as well. Since dropout is replicated, we don't see this but this could cause lots of confusion if we only propagate the mesh.

Reply:
That's a valid concern. I could manually set meshes for all "tensor factory" calls -- only `uniform` is used in the transformer layer. 

I wonder your thoughts on a "good default". For UX, I'm OK with having to set meshes for fusion inputs and outputs. However, having to do for intermediate tensors isn't ideal. In addition to the extra code the user has to write, these tensors might not always be exposed to Thunder. `uniform` happens to be because it's a one-to-one mapping, but a Thunder prim can well be converted to a tensor factory call followed by an elementwise...